server/sonar-server/src/main/java/org/sonar/server/qualitygate/ws/GetByProjectAction.java

```diff
@@ -79,15 +83,20 @@ public void define(WebService.NewController context) {
       .setDescription("Project key")
       .setExampleValue(KEY_PROJECT_EXAMPLE_001)
       .setRequired(true);
+
+    wsSupport.createOrganizationParam(action);
   }
 
   @Override
   public void handle(Request request, Response response) throws Exception {
     try (DbSession dbSession = dbClient.openSession(false)) {
+      OrganizationDto organization = wsSupport.getOrganization(dbSession, request);
       ComponentDto project = componentFinder.getByKey(dbSession, request.mandatoryParam(PARAM_PROJECT));
+      // As ComponentFinder doesn't handle organization yet, we only check here that the project belongs to the organization
+      wsSupport.checkProjectBelongsToOrganization(organization, project);
 
-      if (!userSession.hasComponentPermission(UserRole.USER, project) &&
-        !userSession.hasComponentPermission(UserRole.ADMIN, project)) {
+      if (!userSession.hasComponentPermission(USER, project) &&
```

Comment:
I'm wondering why we are checking that it's not a User, Admin isn't sufficient ?

Reply:
Nope, as the goal is to allow a simple user to access to this WS

Reply:
Ok understood, it may be more lisible to write it without the negation : 
```java
      if (userSession.hasComponentPermission(UserRole.USER, project) ||
        userSession.hasComponentPermission(UserRole.ADMIN, project)) {
```

So it's less brain consuming, a user or an admin can do that.